lib/game.rb

```diff
@@ -0,0 +1,60 @@
+# tracks frames and handles bonus logic
+class Game
+  attr_reader :current_frame, :frames, :score_board
+
+  def initialize(args = {})
+    @frame_class = args[:frame_class] || Frame
+    @score_board = args[:score_board] || ScoreBoard
+    @current_frame = frame_class.new
+    @frames = []
+  end
+
+  def add_roll(pins)
+    raise GameOverError if game_over?
+
+    current_frame.add_roll(pins)
+    add_bonus(pins) unless frames.empty?
+    finish_frame if current_frame.over?
+    score
+  end
+
+  def score
+    score_board.total_score(frames)
+  end
+
+  private
+
+  attr_reader :frame_class
+  attr_writer :current_frame
+
+  def finish_frame
+    frames << current_frame
+    return if game_over?
+
+    self.current_frame = frame_class.new
+    current_frame.final if frames.count == 9
+  end
```

Comment:
Not sure really, just the way I designed it - I chose to keep it in current frame so that I could then add rolls to it there, etc. before pushing it. I suppose I could push it beforehand instead and do the same with `frames.last` instead

Do you think that would be better from a design point of view?